Pass PIPELINE_COLLECTOR_ENABLED env var to Pipeline config

mcranston18 opened the pull request:
On each page refresh, `collectstatic` is run. This makes development extremely slow on OSX due to the way OSX handles shared file systems. For me, each page refresh takes between 10s-20s:

<img width="1219" alt="questions" src="https://user-images.githubusercontent.com/1489696/57036164-7db14180-6c21-11e9-899d-072928c10d94.png">

We can prevent `collectstatic` from being run on every page refresh by passing `PIPELINE_COLLECTOR_ENABLED` to the Pipeline configuration. If that env var is not present, it defaults to `True` which keeps the behaviour consistent with the current dev.

After disabling `PIPELINE_COLLECTOR_ENABLED`, the same page takes 1s instead of 20s:

<img width="1229" alt="question_after" src="https://user-images.githubusercontent.com/1489696/57036282-d7b20700-6c21-11e9-918c-185f4ef3f2d8.png">

akatsoulas commented:
@pmac @jpetto could you review this? I don't have access to OSX to test it.

pmac commented:
No concerns. This should already have been the case! Can't believe we've not done this to this point. My only suggestion would be to default this to `False` when `DEBUG` is on. So change it to `default=not DEBUG` and change the note to indicate how to set it manually if this isn't working for you. It should be fine for everyone being off for dev though.

akatsoulas commented:
@mcranston18 could you open this PR against the master branch?

mcranston18 commented:
@akatsoulas I've updated the base branch to be `master`.

@pmac I've added your suggestion and default the value to `not DEBUG`. One other note, I believe a new dev will have to manually run `collectstatic` on project set up in order to initially compile the css. Is this correct? If so, maybe I should document that as well

pmac commented:
> One other note, I believe a new dev will have to manually run `collectstatic` on project set up in order to initially compile the css. Is this correct? If so, maybe I should document that as well

We should test that. I don't think that's the case since Django should find and load resources from where they live in apps and directories without having to have them in a single static folder. But perhaps pipeline does need that to do whatever compiling it needs to do?

mcranston18 commented:
@pmac I rebuilt the project on this branch and the styles did appear to be compiled using the new settings:

![Screen Shot 2019-05-07 at 11 58 58 AM](https://user-images.githubusercontent.com/1489696/57314516-c8b6d300-70bf-11e9-9c89-6342fc459a0a.png)

mcranston18 commented:
@pmac I finally got around to testing this. It seems that `PIPELINE_COLLECTOR_ENABLED` must be `TRUE` initially for a new developer to set the project up. When `PIPELINE_COLLECTOR_ENABLED` is turned off, the static files aren’t collected which means there won’t be a `static` folder in the root of the project and then you’ll get:

![Screen Shot 2019-05-07 at 12 54 56 PM](https://user-images.githubusercontent.com/1489696/57319193-09b3e500-70ca-11e9-9cff-f596679bedf6.png)

I just updated the code so `PIPELINE_COLLECTOR_ENABLED` defaults back to `True`. If you'd rather, we could leave it to `not DEBUG` but then we'd need to mention in the read me that a new dev must run `collectstatic` when starting the project for the first time.

pmac commented:
I think defaulting to `not DEBUG` and adding the note to the README is the way to go here. Another option is to add a `collectstatic` step to the server startup procedure in the scripts. That might make the dev server take a bit more time to start but it would keep each request snappy.

jpetto commented:
I like the idea of adding a `collectstatic` call to the server startup for local dev.

pmac commented on README.md:
````diff
@@ -55,6 +55,14 @@ instead of `make shell` as the latter does not bind port 8000 which you need to
+By default, the `.less` files do not compile in local development because it impacts load times on OSX. If you are working on `.less` files, you can enable compilation with the following environment variable:
+
+  ```
+  echo "PIPELINE_COLLECTOR_ENABLED=True" >> .env
````
👍

mcranston18 replied:
@pmac Happy to help ! I'll leave it to you to merge this in once ready (I don't have write access to the repo)

pmac replied:
Sounds good. I'll let @akatsoulas have the final look and let him merge it since he's primary on the project. But it all looks good to me! 🍰
